Compilation on old-box Fedora 21

Hello everyone and many thanks for an execellent emulator.
I have encountered a trivial problem (I think) when compiling `unicorn` (release version 0.9) on my box `fedora 21` (yes, it is old but I cannot upgrade it now because of some ABI compatibility problems with libraries of my current project). I have typed `./make.sh` with the edited `config.mk`

> UNICORN_DEBUG ?= no
> UNICORN_ARCHS ?= x86
> UNICORN_STATIC ?= yes
> UNICORN_SHARED ?= yes

then received some linkage errors:

```
  LINK    sample_x86
../libunicorn.so: error: undefined reference to 'pthread_mutexattr_init'
../libunicorn.so: error: undefined reference to 'pthread_mutexattr_settype'
../libunicorn.so: error: undefined reference to 'pthread_mutexattr_destroy'
../libunicorn.so: error: undefined reference to 'pthread_mutex_trylock'
../libunicorn.so: error: undefined reference to 'sem_init'
../libunicorn.so: error: undefined reference to 'sem_destroy'
../libunicorn.so: error: undefined reference to 'sem_post'
../libunicorn.so: error: undefined reference to 'sem_trywait'
../libunicorn.so: error: undefined reference to 'sem_timedwait'
../libunicorn.so: error: undefined reference to 'sem_wait'
../libunicorn.so: error: undefined reference to 'pthread_sigmask'
../libunicorn.so: error: undefined reference to 'pthread_create'
../libunicorn.so: error: undefined reference to 'pthread_join'
../libunicorn.so: error: undefined reference to 'pthread_kill'
collect2: erreur: ld a retourné 1 code d'état d'exécution
Makefile:120: recipe for target 'sample_x86' failed
make[1]: *** [sample_x86] Error 1
make[1]: Leaving directory '/home/bigraph/Public/unicorn-0.9/samples'
Makefile:188: recipe for target 'all' failed
make: *** [all] Error 2
```
It is indeed trivial to fix by changing `LDFLAGS` in `sample/Makefile` into:

`LDFLAGS += -L$(LIBDIR) -l$(LIBNAME) -lpthread`

This is natural to me since `pthread` should be added, but I am not sure it is the same on others OS or versions of Linux.

--- a/make.sh
+++ b/make.sh
@@ -51,7 +51,7 @@ install() {
     rm -rf /usr/lib/libunicorn*
     rm -rf /usr/include/unicorn
     # install into /usr/local
-    PREFIX=/usr/local
+    PREFIX="${PREFIX-/usr/local}"
     ${MAKE} install
   else  # not OSX
     test -d /usr/lib64 && LIBDIRARCH=lib64
@@ -64,7 +64,7 @@ uninstall() {
   if [ "$UNAME" = "Darwin" ]; then
     # find the directory automatically, so we can support both Macport & Brew
     PKGCFGDIR="$(pkg-config --variable pc_path pkg-config | cut -d ':' -f 1)"
-    PREFIX=/usr/local
+    PREFIX="${PREFIX-/usr/local}"
     ${MAKE} uninstall
   else  # not OSX
     test -d /usr/lib64 && LIBDIRARCH=lib64
@@ -80,7 +80,7 @@ fi
 
 if [ -n "`echo "$UNAME" | grep BSD`" ]; then
   MAKE=gmake
-  PREFIX=/usr/local
+  PREFIX="${PREFIX-/usr/local}"
 fi
 
 [ -z "${UNAME}" ] && UNAME=$(uname)

